### PR TITLE
Merge master into dev/repo-refactoring

### PR DESCRIPTION
**Pull request by livarcocc**

### CODE REVIEW

File: TestAssets/DesktopTestProjects/LibWithProjTool2Fx/Lib.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>netstandard1.6;net451</TargetFrameworks>
     <OutputType>Library</OutputType>
-    <AssetTargetFallback Condition=" '$(TargetFramework)' == 'netstandard1.6' ">$(AssetTargetFallback);portable-net45+win8;dnxcore50;netcoreapp2.0</AssetTargetFallback>
+    <AssetTargetFallback Condition=" '$(TargetFramework)' == 'netstandard1.6' ">$(AssetTargetFallback);portable-net45+win8;dnxcore50;netcoreapp2.1</AssetTargetFallback>
```
Comment (tannergooding) [inline TestAssets/DesktopTestProjects/LibWithProjTool2Fx/Lib.csproj]: Do we still need most of these fallback paths?

File: TestAssets/TestPackages/dotnet-prefercliruntime/dotnet-prefercliruntime.csproj [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+    <TargetFramework>netcoreapp2.1</TargetFramework>
     <OutputType>Exe</OutputType>
     <VersionSuffix></VersionSuffix>
   </PropertyGroup>
 
   <ItemGroup>
     <Content Include="prefercliruntime">
       <Pack>true</Pack>
-      <PackagePath>\prefercliruntime</PackagePath>
+      <PackagePath>/</PackagePath>
```
Comment (tannergooding) [inline TestAssets/TestPackages/dotnet-prefercliruntime/dotnet-prefercliruntime.csproj]: Do we actually need this? Can it be blank or `./` instead?

File: TestAssets/TestProjects/AppWithCorruptedLaunchSettings/AppWithCorruptedLaunchSettings.csproj [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.12-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.16.10-x64;centos.7-x64;rhel.7-x64;debian.8-x64;fedora.24-x64;opensuse.42.1-x64</RuntimeIdentifiers>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.12-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.16.10-x64;rhel.6-x64;centos.7-x64;rhel.7-x64;debian.8-x64;fedora.24-x64;opensuse.42.1-x64;linux-musl-x64</RuntimeIdentifiers>
```
Comment (tannergooding) [inline TestAssets/TestProjects/AppWithCorruptedLaunchSettings/AppWithCorruptedLaunchSettings.csproj]: does fedora27 and the other new distro's also need to be in here?

File: build.sh [hunk 4]
```diff
@@ -24,39 +24,31 @@ if [ -z "$HOME" ]; then
     mkdir -p $HOME
 fi
```
Comment (tannergooding) [inline build.sh]: Run the file through shell check please.

File: build/AppHostTemplate.proj [hunk 5]
```diff
@@ -0,0 +1,49 @@
+<Project ToolsVersion="15.0" DefaultTargets="ExtractAppHostToOutput">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.tasks))\dir.tasks" />
+
+  <PropertyGroup>
+    <NativeExecutableExtension Condition=" $(Rid.StartsWith('win'))">.exe</NativeExecutableExtension>
```
Comment (tannergooding) [inline build/AppHostTemplate.proj]: Do we not have a general SDK property for this already?

File: build/AppHostTemplate.proj [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    </ItemGroup>
+
+    <Error Condition="@(NativeRestoredAppHostNETCore->Distinct()->Count()) != 1"
+           Text="Failed to determine the $(_NETCoreDotNetAppHostPackageName) executable in $(AllFileOfRestoredAppHostPackage)" />
+
+    <Copy
+        SourceFiles="@(NativeRestoredAppHostNETCore)"
+        DestinationFolder="$(AppHostTemplatePath)" />
+
+    <Message Text="Copy from @(NativeRestoredAppHostNETCore) to $(AppHostTemplatePath)."
```
Comment (tannergooding) [inline build/AppHostTemplate.proj]: Shouldn't the message come before the copy operation?

**Comment by livarcocc** [comment]
@tannergooding most or all of your comments are actually about changes that happened in master (or some other branch upstream to it) and were not really made by me. While I could go ahead and fix them, I would like to keep this PR just about the merge and the work to get the branch building and producing the appropriate artifacts again.

**Comment by tannergooding** [comment]
@livarcocc, agreed. However, it would be nice to know which we need tracking bugs for and which are things which are "by design".

File: build/package/Layout.targets [hunk 7]
```diff
@@ -89,6 +80,8 @@
           Inputs="@(LayoutDefinition -> '%(InputFiles)')"
           Outputs="@(LayoutDefinition -> '%(OutputFiles)')" >
 
+    <Message Text="DEBUG: %(LayoutDefinition.OutputFiles)" Importance="High" />
```
Comment (livarcocc) [inline build/package/Layout.targets]: Need to remove this console.writeline.

**Comment by livarcocc** [comment]
@tannergooding we absolutely should file issues for things that need to be fixed. However, if they are working, we need good reasons to change them. Also, for simple things, I would say just go ahead and get a PR up. It would be more expedient.

**Comment by livarcocc** [comment]
Runtime failures.

@dotnet-bot Test Debian8.2 x64 Debug Build
@dotnet-bot Test Ubuntu x64 Release Build

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

**Pull request merged**